src/sonic_ax_impl/mibs/ieee802_1ab.py

```diff
@@ -5,25 +5,160 @@
+
+    def get_next(self, sub_id):
+        """
+        :param sub_id: The 1-based sub-identifier query.
+        :return: the next sub id.
+        """
+        right = bisect_right(self.if_range, sub_id)
+        if right == len(self.if_range):
+            return None
+        return self.if_range[right]
+
+    def update_data(self):
+        """
+        Subclass update data routine. Updates available local port data.
+        """
+        # establish connection to application database.
+        self.db_conn.connect(mibs.APPL_DB)
+        self.if_range = []
+        # get local port kvs from APP_BD's PORT_TABLE
+        self.loc_port_data = {}
+        for if_oid, if_name in self.oid_name_map.items():
+            loc_port_kvs = self.db_conn.get_all(mibs.APPL_DB, mibs.if_entry_table(if_name))
+            if not loc_port_kvs:
+                continue
+            self.if_range.append((if_oid, ))
+            self.loc_port_data.update({if_name: loc_port_kvs})
+        self.if_range.sort()
+        if not self.loc_port_data:
+            logger.warning("0 - b'PORT_TABLE' is empty. No local port information could be retrieved.")
+
+    def local_port_num(self, sub_id):
+        if len(sub_id) <= 0:
+            return None
+        sub_id = sub_id[0]
+        if sub_id not in self.oid_name_map:
+            return None
+        return int(sub_id)
+
+    def local_port_id(self, sub_id):
+        if len(sub_id) <= 0:
+            return None
+        sub_id = sub_id[0]
+        if sub_id not in self.oid_name_map:
+            return None
+        if_name = self.oid_name_map[sub_id]
+        if if_name not in self.loc_port_data:
+            # no LLDP data for this interface--we won't report the local interface
+            return None
+        return self.if_alias_map[if_name]
+
+    def port_table_lookup(self, sub_id, table_name):
+        if len(sub_id) <= 0:
+            return None
+        sub_id = sub_id[0]
+        if sub_id not in self.oid_name_map:
+            return None
+        if_name = self.oid_name_map[sub_id]
+        if if_name not in self.loc_port_data:
+            # no data for this interface
+            return None
+        counters = self.loc_port_data[if_name]
+        _table_name = bytes(getattr(table_name, 'name', table_name), 'utf-8')
+        try:
+            return counters[_table_name]
+        except KeyError as e:
+            mibs.logger.warning(" 0 - b'PORT_TABLE' missing attribute '{}'.".format(e))
+            return None
+
+
+class LLDPLocalSystemDataUpdater(MIBUpdater):
+    def __init__(self):
+        super().__init__()
+
+        self.db_conn = mibs.init_db()
+        self.loc_chassis_data = {}
+
+    def update_data(self):
+        """
+        Subclass update data routine.
+        """
+        # establish connection to application database.
+        self.db_conn.connect(mibs.APPL_DB)
+        self.loc_chassis_data = self.db_conn.get_all(mibs.APPL_DB, mibs.LOC_CHASSIS_TABLE)
```

Comment:
>get_all [](start = 45, length = 7)

Again, do we need to update so frequently?

Reply:
You are right,  here the data is basically static, I will move this to reinit(), so it will be updated once a minute.